### ISSUE TITLE
External Modules don't output correctly for 'global' type.

### ISSUE THREAD

**Issue by marcusdarmstrong** [webpack/webpack#6933]
https://github.com/webpack/webpack/blob/0e404b8683d79052e3a6b42743bd3061f17a5a5d/lib/ExternalModule.js#L116

Hi there, in attempting to get our server side bundles set up properly, I think I discovered a bug introduced alongside the addition of configurable global objects (https://github.com/webpack/webpack/commit/d95f93f8cb875d4184c38ef3548aaca2f9ac8175)

 As of Webpack 4.4.1, this is the output I get back from the 'global' option:
```js
/***/ (function(module, exports) {

(function() { module.exports = global["self"]; }());

/***/ })
```

I believe the intent of the commit was:
```js
return this.getSourceForGlobalVariableExternal(request, runtime.outputOptions.globalObject);
```

Which would lookup the module by name in the provided global object.

**Do you want to request a *feature* or report a *bug*?** Bug
 

**What is the current behavior?** the source() of an ExternalModule with type 'global' yields the above snippet.

**If the current behavior is a bug, please provide the steps to reproduce.** Call .source() on an ExternalModule with type 'global'.

**What is the expected behavior?** I believe the intent is to lookup the module by name in the configured global, rather than look up the configured global in global scope (which is kinda what the current code does?)

**If this is a feature request, what is motivation or use case for changing the behavior?** N/A

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.** Webpack 4.4.1, but it's the same in master.

**Comment by montogeek** [comment]
It looks like you just deleted our lovely crafted [issue template](https://raw.githubusercontent.com/webpack/webpack/master/.github/ISSUE_TEMPLATE.md). It was there for good reasons. Please help us solving your issue by answering the questions asked in this template.Please either update the issue with the template and reopen, or open a new issue.

**Comment by marcusdarmstrong** [comment]
Sorry about that, I didn't delete the issue template. I opened the issue from the github source browser, so it never gave me a template in the first place.

**Comment by sokra** [comment]
Yep, that's wrong, send a PR

**Comment by edmorley** [comment]
I believe this can be closed now, since #7038 was merged.

**Comment by evilebottnawi** [comment]
Solved in #7038
